pkg/client/clientcmd/loader.go

```diff
@@ -134,8 +134,8 @@ func (rules *ClientConfigLoadingRules) Load() (*clientcmdapi.Config, error) {
 	// since values are overwritten, but maps values are not, we can merge the non-map config on top of the map config and
 	// get the values we expect.
 	config := clientcmdapi.NewConfig()
-	mergo.Merge(config, mapConfig)
-	mergo.Merge(config, nonMapConfig)
+	mergo.MergeWithOverwrite(config, mapConfig)
```

Comment:
Previously, map values were only overwritten if no key was set in the destination map before.  With this change, the destination map value will be overwritten.  I think the new semantic could be used to make this function easier to ready by eliminating the distinction between `mapConfig` and `nonMapConfig`, but as it is now isn't the `mapConfig` essentially stomped?